File: src/main/java/com/itsci/mju/maebanjumpen/partyrole/service/HirerService.java
package com.itsci.mju.maebanjumpen.partyrole.service;

import com.itsci.mju.maebanjumpen.partyrole.dto.HirerDTO;
import com.itsci.mju.maebanjumpen.exception.HirerNotFoundException;
import com.itsci.mju.maebanjumpen.exception.InsufficientBalanceException;

import java.util.List;

public interface HirerService {
    HirerDTO saveHirer(HirerDTO hirerDto);
    HirerDTO getHirerById(int id);
    List<HirerDTO> getAllHirers();
    HirerDTO updateHirer(int id, HirerDTO hirerDto);
    void deleteHirer(int id);

    void deductBalance(Integer hirerId, Double amount) throws InsufficientBalanceException, HirerNotFoundException;
    void addBalance(Integer hirerId, Double amount) throws HirerNotFoundException;


}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/service/HirerService.kt
package com.itsci.mju.maebanjumpen.partyrole.service

import com.itsci.mju.maebanjumpen.partyrole.dto.HirerDTO

interface HirerService {
    fun saveHirer(hirerDto: HirerDTO): HirerDTO
    fun getHirerById(id: Int): HirerDTO
    fun getAllHirers(): List<HirerDTO>
    fun updateHirer(id: Int, hirerDto: HirerDTO): HirerDTO
    fun deleteHirer(id: Int)
    fun deductBalance(hirerId: Int, amount: Double)
    fun addBalance(hirerId: Int, amount: Double)
}

